pkg/manager/internal/webhooks/customresourcediscovery_webhook.go

```diff
@@ -20,25 +20,35 @@ import (
 	"context"
 	"fmt"
 	"net/http"
+	"strings"
 
 	"github.com/go-logr/logr"
 	adminv1beta1 "k8s.io/api/admission/v1beta1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
 
 	corev1alpha1 "github.com/kubermatic/kubecarrier/pkg/apis/core/v1alpha1"
+	"github.com/kubermatic/kubecarrier/pkg/internal/util"
 )
 
 // CustomResourceDiscoveryWebhookHandler handles mutating/validating of CustomResourceDiscoveries.
 type CustomResourceDiscoveryWebhookHandler struct {
 	decoder *admission.Decoder
 	Log     logr.Logger
+	client.Client
+	Scheme *runtime.Scheme
 }
 
 var _ admission.Handler = (*CustomResourceDiscoveryWebhookHandler)(nil)
 
-// +kubebuilder:webhook:path=/validate-kubecarrier-io-v1alpha1-customresourcediscovery,mutating=false,failurePolicy=fail,groups=kubecarrier.io,resources=customresourcediscoveries,verbs=update,versions=v1alpha1,name=vcustomresourcediscovery.kubecarrier.io
+// +kubebuilder:webhook:path=/validate-kubecarrier-io-v1alpha1-customresourcediscovery,mutating=false,failurePolicy=fail,groups=kubecarrier.io,resources=customresourcediscoveries,verbs=update;delete,versions=v1alpha1,name=vcustomresourcediscovery.kubecarrier.io
 
-// Handle is the function to handle create/update requests of CustomResourceDiscoveries.
+// Handle is the function to handle update/delete requests of CustomResourceDiscovery objects.
 func (r *CustomResourceDiscoveryWebhookHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
 	obj := &corev1alpha1.CustomResourceDiscovery{}
```

Comment:
IIRC delete admission hook doesn't have current obj setup, thus this would error out. Move this to the `UPDATE` section.